src/mmw/apps/modeling/mapshed.py

```diff
@@ -57,6 +57,9 @@ def nearest_weather_stations(geom, n=NUM_WEATHER_STATIONS):
     with connection.cursor() as cursor:
         cursor.execute(sql, [geom.wkt, n])
 
+        if cursor.rowcount == 0:
+            raise Exception("No weather stations found.")
+
```

Comment:
I don't think there can ever be 0, which is why this is such an exceptional case. This would probably indicate a problem in the dataset itself